doc/auth.md

```diff
@@ -230,12 +230,581 @@ Response Payload :
+3. Send email reset password link to user.
+
+
+#### Validation
+- email: required, using email pattern, and not empty
+
+
+
+### Scenario Test
+
+#### Case : Negative Case 1
+
+Request Payload : empty
+
+Response HTTP Status Code : 400
+
+Response Payload :
+```
+{
+    "status_code": "cdc-400",
+    "status_message": "email is required",
+    "data": null
+}
+```
+
+#### Case : Negative Case 2
+
+Request Payload :
+```
+{}
+```
+
+Response HTTP Status Code : 400
+
+Response Payload :
+```
+{
+    "status_code": "cdc-400",
+    "status_message": "email is required",
+    "data": null
+}
+```
+
+#### Case : Negative Case 3
+
+Request payload :
+```
+{
+    "email": ""
+}
+```
+
+Response HTTP Status Code : 400
+
+Response Payload :
+```
+{
+    "status_code": "cdc-400",
+    "status_message": "email is empty",
+    "data": null
+}
+```
+
+#### Case : Negative Case 4
+
+Request Payload :
+```
+{
+    "email": "asal"
+}
+```
+
+Response HTTP Status Code : 400
+
+Response Payload :
+```
+{
+    "status_code": "cdc-400",
+    "status_message": "email format is invalid",
+    "data": null
+}
+```
+
+
+#### Case : Negative Case 5
+
+Request Payload
+```
+{
+    "email": "asal@gmail.com"
+}
+```
+
+Response HTTP Status Code : 404
+
+Response Payload
+```
+{
+    "status_code": "cdc-404",
+    "status_message": "email not found",
+    "data": null
+}
+```
+
+#### Case : Positive Case
+
+Request Payload :
+```
+{
+    "email": "jacky@gmail.com"
+}
+```
+
+Response HTTP Status Code : 200
+
+Response Payload :
+```
+{
+    "status_code": "CDC-200",
+    "status_message": "Password recovery was sended to your email. please checked your email. ",
+    "data": null
+}
+```
+
 ## <a name="change-password"></a>Change Password
 
 ### Endpoint
 POST /change-password
 
+For change your password you need to input your old password and your new password
+
+
+### Headers
+Key | Value 
+--- | ---
+Content-Type | application/json
+Accept | application/json
+
+### Request Payloads
+Name | Type | Example Value
+--- | --- | ---
+old_password | string | jakaRTa!2020
+new_password | string | jakaRTa!2021
+re_new_password | string | jakaRTa!2021
+```
+{
+    "old_password": "jakaRTa!2020",
+    "new_password": "jakaRTa!2021",
+    "re_new_password": "jakaRTa!2021",
+}
+```
+### Response Payloads
+HTTP Code | Status | Description
+--- | --- | ---
+400 | Bad Request | Bad request payload  
+404 | Not Found | User not found in database  
+500 | Internal Server Error | some un-handle error in server 
+200 | Success change password | Success change password
+```
+{
+    "status_code": "CDC-400",
+    "status_message": "Bad Request",
+    "data": null
+}
+```
+
+```
+{
+    "status_code": "CDC-200",
+    "status_message": "Success change password.",
+    "data": null
+}
+```
+
+### Logic
+
+1. Validate old password, new password input.
+2. matching new password with re new password.
+3. check if old password input is same like on database.
```

Comment:
- check if username/email/user_id exist in database and get password from database based on username/email/user_id
- compare old password from request with password from database